src/main/java/com/google/sps/utility/ServletUtility.java

```diff
@@ -27,53 +28,69 @@
   private ServletUtility() {}
 
   /**
-   * Helper method to get a cookie from an HttpServletRequest
+   * Helper method to get a cookie from an HttpServletRequest. Use hasCookie to check if cookie
+   * exists before using this.
    *
    * @param request HttpServletRequest that contains desired cookie
    * @param cookieName name of desired cookie. Case sensitive
-   * @return Cookie if found, null if not found or if duplicates present
+   * @return Cookie object that corresponds to the cookieName
+   * @throws CookieParseException if a cookie cannot be found or if duplicates present
    */
-  public static Cookie getCookie(HttpServletRequest request, String cookieName) {
+  public static Cookie getCookie(HttpServletRequest request, String cookieName)
+      throws CookieParseException {
     List<Cookie> cookies =
         Arrays.stream(request.getCookies())
             .filter((Cookie c) -> c.getName().equals(cookieName))
             .collect(Collectors.toList());
 
     if (cookies.isEmpty()) {
-      System.out.println("Cookie not found");
-      return null;
+      throw new CookieParseException("Cookie not found");
     }
 
     // If more than one cookies are found, it is ambiguous as to which one to return.
     // This is unexpected - duplicate cookies are usually blocked by the browser (especially
     // when they are user set).
     if (cookies.size() > 1) {
-      System.out.printf("%d duplicate cookies with name %s\n", cookies.size(), cookieName);
-      cookies.forEach(c -> System.out.println(c.getValue()));
-
-      return null;
+      String errorMessage =
+          String.format("%d duplicate cookies with name %s\n", cookies.size(), cookieName);
+      errorMessage += cookies.stream().map(Cookie::getValue).reduce((a, b) -> a + "\n" + b);
+      throw new CookieParseException(errorMessage);
     }
 
     return cookies.get(0);
   }
 
   /**
-   * Helper function to create an authorization header for an outgoing HTTP request Does NOT check
-   * if the access token provides correct permissions for the request Does NOT check if the
-   * userToken was valid. Use verifyUserToken to verify userId validity
+   * Checks if a cookie by the specified name is present in the request
+   *
+   * @param request HTTP request from client
+   * @param cookieName name of desired cookie. Case sensitive
+   * @return true if present (duplicates will also return true), false otherwise
+   */
+  public static boolean hasCookie(HttpServletRequest request, String cookieName) {
+    List<Cookie> cookies =
+        Arrays.stream(request.getCookies())
+            .filter((Cookie c) -> c.getName().equals(cookieName))
+            .collect(Collectors.toList());
+
+    return cookies.size() > 0;
```

Comment:
extreme nit: Here's a [debate](https://stackoverflow.com/questions/9341740/ifliststr-size-0-versus-ifliststr-isempty/9341835) between using `.size() > 0` and `.isEmpty()`. Don't think it really matters though.

Reply:
Fixed just to keep consistent. I don't know why I keep using .size() instead of .isEmpty() - force of habit I guess!